language/move-vm/types/src/gas.rs

```diff
@@ -1,47 +1,369 @@
+
+    fn charge_move_loc(&mut self, val: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_store_loc(&mut self, val: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_pack(
+        &mut self,
+        is_generic: bool,
+        args: impl ExactSizeIterator<Item = impl ValueView>,
+    ) -> PartialVMResult<()>;
+
+    fn charge_unpack(
+        &mut self,
+        is_generic: bool,
+        args: impl ExactSizeIterator<Item = impl ValueView>,
+    ) -> PartialVMResult<()>;
+
+    fn charge_read_ref(&mut self, ref_val: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_write_ref(&mut self, val: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_eq(&mut self, lhs: impl ValueView, rhs: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_neq(&mut self, lhs: impl ValueView, rhs: impl ValueView) -> PartialVMResult<()>;
+
+    fn charge_borrow_global(
+        &mut self,
+        is_mut: bool,
+        is_generic: bool,
+        ty: impl TypeView,
+        is_success: bool,
+    ) -> PartialVMResult<()>;
+
+    fn charge_exists(
+        &mut self,
+        is_generic: bool,
+        ty: impl TypeView,
+        // TODO(Gas): see if we can get rid of this param
+        exists: bool,
+    ) -> PartialVMResult<()>;
+
+    fn charge_move_from(
+        &mut self,
+        is_generic: bool,
+        ty: impl TypeView,
+        val: Option<impl ValueView>,
+    ) -> PartialVMResult<()>;
+
+    fn charge_move_to(
+        &mut self,
+        is_generic: bool,
+        ty: impl TypeView,
+        val: impl ValueView,
+        is_success: bool,
+    ) -> PartialVMResult<()>;
 
-    /// Charge an instruction over data with a given size and fail if not enough gas units are left.
-    fn charge_instr_with_size(
+    fn charge_vec_pack<'a>(
         &mut self,
-        opcode: Opcodes,
-        size: AbstractMemorySize<GasCarrier>,
+        ty: impl TypeView + 'a,
+        args: impl ExactSizeIterator<Item = impl ValueView>,
     ) -> PartialVMResult<()>;
 
-    /// Charge a given amount in the unit of measurement native to the GasMeter implementation.
+    fn charge_vec_len(&mut self, ty: impl TypeView) -> PartialVMResult<()>;
+
+    fn charge_vec_borrow(
+        &mut self,
+        is_mut: bool,
+        ty: impl TypeView,
+        is_success: bool,
+    ) -> PartialVMResult<()>;
+
+    fn charge_vec_push_back(
+        &mut self,
+        ty: impl TypeView,
+        val: impl ValueView,
+    ) -> PartialVMResult<()>;
+
+    fn charge_vec_pop_back(
+        &mut self,
+        ty: impl TypeView,
+        val: Option<impl ValueView>,
+    ) -> PartialVMResult<()>;
+
+    // TODO(Gas): Expose the elements
+    fn charge_vec_unpack(
+        &mut self,
+        ty: impl TypeView,
+        expect_num_elements: NumArgs,
+    ) -> PartialVMResult<()>;
+
+    // TODO(Gas): Expose the two elements
+    fn charge_vec_swap(&mut self, ty: impl TypeView) -> PartialVMResult<()>;
```

Comment:
Can you make an issue for these TODOs and others?

Reply:
Will do!